INTEGRATING.md

```diff
@@ -1,36 +1,171 @@
-## Integrating Grafana agent
+# Integrating Grafana agent
 
-Grafana agent charm integrates with:
+Grafana Agent Charmed Operator can be integrated with other charms through the following relations.
 
 
-- Any charm that supports the `loki_push_api` interface. You can integrate it using the Loki Charm library. The documentation of the charm library, available through [Charmhub](https://charmhub.io/loki-k8s) provides further details. The information exchanged through the `loki_push` interface can be broken down into two parts:
+## Provides
 
-  - Loki charm provides an [endpoint URL](https://grafana.com/docs/loki/latest/api/#post-lokiapiv1push) to receive log from [Loki clients](https://grafana.com/docs/loki/latest/clients/) that relates with this charm.
+### Logging provider
+
+```yaml
+  logging-provider:
+    interface: loki_push_api
+```
+
+Grafana Agent Charmed Operator may receive logs from any charm that supports the [`loki_push_api`](https://charmhub.io/loki-k8s/libraries/loki_push_api) relation interface.
+
+Besides, the `logging-provider` relation can inject and setup `Promtail` into the workload container of the consumer side of the relation.
+
+The information exchanged through the `loki_push_api` interface can be broken down into two parts:
+
+  - Grafana Agent charm provides an [endpoint URL](https://grafana.com/docs/loki/latest/api/#post-lokiapiv1push) to receive log from [Loki clients](https://grafana.com/docs/loki/latest/clients/) that relates with this charm.
+  - Grafana Agent charm may also receive alert rules which tell when to raise alerts. These rules are read from a directory named `loki_alert_rules`, if present at the top level, within the client charm's source (`src`) directory.
 
-  - Loki may receive alert rules which tell when to raise alerts. These rules are read from a directory named `loki_alert_rules`, if present at the top level, within the client charm's source (`src`) directory.
 
+For instance let's say that we have a [Zinc Charmed Operator](https://charmhub.io/zinc-k8s) that implements the other side (`requires`) of the relation.
+After deploying this charm, we can relate `Grafana Agent` and `Zinc` through `loki_push_api` relation interface:
```

Comment:
Need to clearly differentiate between the charm's function as provider and as requirer of the same relation, but how about:

> To have a charm send logs to loki, you can use one of the "consumer" object offered by the `loki_push_api`.
> 
> First, specify the relation in the charm's metadata.yaml:
> 
> ```yaml
> requires:
>   logging:
>     interface: loki_push_api
> ```
> 
> Next, obtain the library from charmhub:
> 
> ```shell
> charmcraft fetch-lib charms.loki_k8s.v0.loki_push_api
> ```
> 
> Then, use the charm library in the charm:
> 
> ```python
> self._logging = LogProxyConsumer(...)
> # or
> self._logging = LokiPushApiConsumer(...)
> ```
> 
> Finally, relate the app (e.g. zinc) to loki:

Reply:
I like it!